src/main/java/duke/command/DeadlineTaskCommand.java

```diff
@@ -0,0 +1,38 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.TaskList;
+import duke.ui.Errors;
+import duke.ui.Ui;
+
+import java.util.HashMap;
+
+public class DeadlineTaskCommand extends Command {
+    private String description;
+    private String by;
+
+    public DeadlineTaskCommand(String[] command) throws DukeException {
+        if (command.length < 2) {
+            throw new DukeException(Errors.MISSING_DESCRIPTION.MESSAGE);
+        }
+
+        HashMap<String, String> args = Parser.parseArguments(command[1],
+                new String[]{command[0], Deadline.DELIMITER_BY});
```

Comment:
Perhaps this could be split up to be clearer? It is not obvious how the arguments are split here.